File: algorithms/001.Two Sum/Solution.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        int l = nums.length;
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < l; i++) {
          int n = nums[i];
          int diff = target - n;
          if (map.containsKey(diff)) {
            return new int[]{map.get(diff), i};
          }
          map.put(n, i);
        }
        return new int[]{0, 0};
    }
}

File: algorithms/001.Two Sum/solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  hash = {}
  for i in 0..nums.length
    n = nums[i]
    diff = target - n
    t = hash[diff]
    if t != nil
      return [t, i]
    end
    hash[n] = i
  end
end
